src/tallies/tally.cpp

```diff
@@ -1220,6 +1220,30 @@ openmc_tally_set_active(int32_t index, bool active)
   return 0;
 }
 
+extern "C" int
+openmc_tally_get_writeable(int32_t index, bool* writeable)
+{
+  if (index < 0 || index >= model::tallies.size()) {
+    set_errmsg("Index in tallies array is out of bounds.");
+    return OPENMC_E_OUT_OF_BOUNDS;
+  }
+  *writeable = model::tallies[index]->writeable_;
```

Comment:
I would add a C++ getter/setter for the writable property and then use that in the C API here.